src/mail-app/mail/model/MailModel.ts

```diff
@@ -432,6 +433,11 @@ export class MailModel {
 		return !this.logins.isEnabled(FeatureType.DisableMailExport)
 	}
 
+	async loadAndMarkMails(mails: () => Promise<readonly Mail[]>, unread: boolean): Promise<void> {
```

Comment:
`load` methods seems like they should not be on the `MailModel`. I know you are probably trying to reduce duplication but we should think about it again I guess.